onnx/defs/tensor/defs.cc

```diff
@@ -1319,4 +1319,22 @@ ONNX_OPERATOR_SET_SCHEMA(
             }
           }
         }));
+
+ONNX_OPERATOR_SET_SCHEMA(
+    IsNaN,
+    9,
+    OpSchema()
+        .SetDoc(R"DOC(Returns which elements of the input are NaN.)DOC")
+      .Input(0, "X", "input", "T1")
+      .Output(0, "Y", "output", "T2")
+      .TypeConstraint(
+          "T1",
+          {"tensor(float16)","tensor(float)","tensor(double)"},
```

Comment:
inputs are limited to float/double?

Reply:
Because NaN does not apply to integral types. (Or, are you asking about float16?)

Reply:
Thinking more about it. The op returns a boolean tensor if a value is NAN or not. So even if we have integers we'll simply return false for them. So we don't necessarily have to constrain to only floating points. It probably make sense to support all numeric types. Thoughts?

Reply:
But why? It seems strange to support it for int, if it is just going to return false always. Is there some use case? Otherwise, it seems to suggest that there is a way to represent NaN in the int type, which is likely to cause confusion. The real issue lies in the original (preceding) int ops that would have produced a NaN (if it could be represented), may be via overflow or divide-by-zero … so a proper solution would require extending the int ops (like multiplication or division) to return a value to indicate that this happened.

Reply:
Ok, let's keep it float only.